api/maven-api-core/src/main/java/org/apache/maven/api/services/ChecksumAlgorithmService.java

```diff
@@ -0,0 +1,87 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.api.services;
+
+import java.io.*;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.maven.api.Service;
+import org.apache.maven.api.annotations.Experimental;
+import org.apache.maven.api.annotations.Nonnull;
+
+/**
+ * Checksum algorithms service.
+ *
+ * @since 4.0.0
+ */
+@Experimental
+public interface ChecksumAlgorithmService extends Service {
+    /**
+     * Returns {@link ChecksumAlgorithm} for given algorithm name, or throws if algorithm not supported.
+     *
+     * @throws IllegalArgumentException if asked algorithm name is not supported.
+     */
+    @Nonnull
+    ChecksumAlgorithm select(@Nonnull String algorithmName);
+
+    /**
+     * Returns a list of {@link ChecksumAlgorithm} in same order as algorithm names are ordered, or throws if any of the
+     * algorithm name is not supported. The returned list has equal count of elements as passed in collection of names,
+     * and if names contains duplicated elements, the returned list of algorithms will have duplicates as well.
+     *
+     * @throws IllegalArgumentException if any asked algorithm name is not supported.
+     * @throws NullPointerException if passed in list of names is {@code null}.
+     */
+    @Nonnull
+    Collection<ChecksumAlgorithm> selectList(@Nonnull Collection<String> algorithmNames);
+
+    /**
+     * Returns immutable collection of all supported algorithm names.
+     */
+    @Nonnull
+    Collection<String> getChecksumAlgorithmNames();
+
+    /**
+     * Calculates checksums for specified data.
+     *
+     * @param data        The content for which to calculate checksums, must not be {@code null}.
+     * @param algorithms  The checksum algorithm factories to use, must not be {@code null}.
+     * @return The calculated checksums, indexed by algorithm name, or the exception that occurred while trying to
+     * calculate it, never {@code null}.
+     * @throws IOException In case of any problem.
+     */
+    @Nonnull
+    Map<String, String> calculate(@Nonnull byte[] data, @Nonnull Collection<ChecksumAlgorithm> algorithms)
+            throws IOException;
+
+    /**
+     * Calculates checksums for specified file.
+     *
+     * @param file        The file for which to calculate checksums, must not be {@code null}.
+     * @param algorithms  The checksum algorithm factories to use, must not be {@code null}.
+     * @return The calculated checksums, indexed by algorithm name, or the exception that occurred while trying to
+     * calculate it, never {@code null}.
+     * @throws IOException In case of any problem.
+     */
+    @Nonnull
+    Map<String, String> calculate(@Nonnull Path file, @Nonnull Collection<ChecksumAlgorithm> algorithms)
```

Comment:
For ease of use ,we're missing `calculate(InputStream, Collection<ChecksumAlgorithm>)` imho.